### ISSUE TITLE
Incorrect offset of const buffer in D3D11 HLSL

### ISSUE THREAD

**Issue by darksylinc** [baldurk/renderdoc#1576]
## Description

We have a shader that defines the following struct:
```

struct Light
{
	float3 position;
	uint lightMask; // <--- This one wrecks off RenderDoc
	float4 diffuse;
	float3 specular;

	float3 attenuation;
	float4 spotDirection;
	float4 spotParams;
};

cbuffer PassBuffer : register(b0)
{
struct _PassBuffer
{
    ...
    Light lights[3];
};
}passBuf;
```

passBuf.lights[0] is recognized fine by RenderDoc.
However passBuf.lights[1] and passBuf.lights[2] are not, as if there was some missing padding

Our OpenGL version works fine.

I can tell this is a RenderDoc bug because:

1. To the best of my knowledge according to D3D11 packing rules, my code is correct
2. The rendered output looks 100% correct, however by looking how off the values end up, the spot lights in the scene should be completely screwed (position, direction, NaN and negative colours, all should be wrong)

## Steps to reproduce

1. Open the uploaded rdc capture
2. Select almost any draw call from the final target to the window
3. See the pass buffer (see screenshot)

## Environment

* RenderDoc version: 1.5
* Operating System: Windows 10
* API: Direct3D 11

Uploads will follow very soon

**Comment by darksylinc** [comment]
Capture:
[Incorrect.rdc.zip](https://github.com/baldurk/renderdoc/files/3804908/Incorrect.rdc.zip)

HLSL Shader code:
[ShaderCode.hlsl.txt](https://github.com/baldurk/renderdoc/files/3804912/ShaderCode.hlsl.txt)

Generated asm from fxc.exe to compare the offsets:
[OutputShaderFromFXC.asm.txt](https://github.com/baldurk/renderdoc/files/3804913/OutputShaderFromFXC.asm.txt)

Exact step where things go wrong:
![RenderDoc](https://user-images.githubusercontent.com/3395130/68137372-a1717f00-ff05-11e9-9178-10442b5e5d75.png)

**Comment by darksylinc** [comment]
I forgot one more detail.

This bug only happens when we enable fine grained light masks. When that's enabled, we define our shader as:
```
	float3 position;
	uint lightMask;
```

However when that feature is disabled, we define the struct as:
```
	float4 position; //float4, not float3
```

and RenderDoc prints the output correctly

**Comment by baldurk** [comment]
Thanks for the report. Can you test out a recent nightly build? I think this is the bug that was fixed in 273a87e6 as reported by someone else coincidentally recently. At least when I load the capture you uploaded in a recent nightly, the constant buffer seems to have correct values.

**Comment by darksylinc** [comment]
Talk about FAST!

Damn, you do deserve Ogre 2.1's codename named after you.

Yeah, the offsets are correct in latest Nightly
